src/controller.py

```diff
@@ -19,66 +20,48 @@
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
-
-def handle_message(message, content):
-    action = content.get("action")
+def handle_drpc_request(drpc_request, connection_id):
     handler = {
-        "request_nonce": handle_request_nonce,
-        "challenge_response": handle_challenge_response,
-    }.get(action, handle_default)
-
-    return handler(message["connection_id"], content)
-
-
-def report_failure(connection_id):
-    message_templates_path = os.getenv("MESSAGE_TEMPLATES_PATH")
-    with open(os.path.join(message_templates_path, "report_failure.json"), "r") as f:
-        report_failure = json.load(f)
-
-    json_str = json.dumps(report_failure)
-    base64_str = base64.b64encode(json_str.encode("utf-8")).decode("utf-8")
+        "request_nonce": handle_drpc_request_nonce,
+    }.get(drpc_request["method"], handle_drpc_default)
 
-    logger.info(f"sending report failure message to {connection_id}")
-
-    send_message(connection_id, base64_str)
+    return handler(drpc_request, connection_id)
 
+def handle_drpc_default(drpc_request, connection_id):
+    return {"jsonrpc": "2.0", "error":{"code": -32601, "message": "method not found"}, "id": drpc_request["id"]}
 
-def handle_request_nonce(connection_id, content):
-    logger.info("handle_request_nonce")
-    connection = get_connection(connection_id)
-    logger.info(f"fetched connection, id = {connection_id}")
-
-    if connection is None or connection["rfc23_state"] != "completed":
-        logger.info(f"connection not completed, id = {connection_id}")
-        return
+def handle_drpc_request_nonce(drpc_request, connection_id):
 
     message_templates_path = os.getenv("MESSAGE_TEMPLATES_PATH")
-    with open(
-        os.path.join(message_templates_path, "request_attestation.json"), "r"
-    ) as f:
-        request_attestation = json.load(f)
-
     nonce = secrets.token_hex(16)
+    request_attestation = {
+        "jsonrpc": "2.0",
+        "method": "request_attestation",
+        "params": {"nonce": nonce},
+        "id": random.randint(0, 1000000),
+    }
     # cache nonce with connection id as key, allow it to expire
     # after n seconds
     redis_instance.setex(connection_id, auto_expire_nonce, nonce)
 
-    request_attestation["nonce"] = nonce
-    json_str = json.dumps(request_attestation)
-    base64_str = base64.b64encode(json_str.encode("utf-8")).decode("utf-8")
-
-    logger.info(f"sending request attestation message to {connection_id}")
+    send_drpc_request(connection_id, request_attestation)
 
-    send_message(connection_id, base64_str)
+    return {}
```

Comment:
Why this return?

Reply:
Because it's a drpc notification response. For the drpc protocol, when a client sends a drpc notification (same as a drpc request but the id is `null`), the server must respond with an empty json object `{}`. 

https://github.com/hyperledger/aries-rfcs/pull/804/files#diff-b18d7b2a9c7c7719f831100c571f40ed66a1f93fa03905da6ee544fe170b4c48R271